sky/services/ns_net/url_loader_impl.mm

```diff
@@ -32,6 +32,17 @@ - (void)connection:(NSURLConnection*)connection
   _response->status_code = response.statusCode;
   _response->url =
       mojo::String(self.originalRequest.URL.absoluteString.UTF8String);
+  NSUInteger headerCount = response.allHeaderFields.count;
+  if (headerCount > 0) {
+    _response->headers = mojo::Array<mojo::HttpHeaderPtr>::New(0);
```

Comment:
Because you know the count, you can allocate the correct amount of space here and avoid resizes.